qiskit/circuit/gate.py

```diff
@@ -24,27 +26,67 @@
 class Gate(Instruction):
     """Unitary gate."""
 
-    def __init__(self, name, num_qubits, params, label=None):
+    def __init__(self, name, num_qubits, params, phase_angle=0, label=None):
         """Create a new gate.
 
         Args:
             name (str): the Qobj name of the gate
             num_qubits (int): the number of qubits the gate acts on.
             params (list): a list of parameters.
-            label (str or None): An optional label for the gate [Default: None]
+            phase_angle (float): set the exponent of the gate phase (Default: 0).
+            label (str or None): An optional label for the gate (Default: None).
         """
         self._label = label
         self.definition = None
+        self._phase_angle = 0
+        if phase_angle:
+            self.phase_angle = phase_angle
         super().__init__(name, num_qubits, 0, params)
 
+    @property
+    def phase(self):
+        """Return the phase of the gate."""
+        return cmath.exp(1j * self.phase_angle) if self.phase_angle else 1
+
+    @phase.setter
+    def phase(self, value):
+        """Set the phase of the gate."""
+        scale, angle = cmath.polar(value)
+        if scale != 1:
+            raise ValueError("Phase value must have absolute value of 1.")
+        self.phase_angle = angle
+
+    @property
+    def phase_angle(self):
+        """Return the phase angle of the gate."""
+        return self._phase_angle
+
+    @phase_angle.setter
+    def phase_angle(self, angle):
+        """Set the phase angle of the gate."""
+        # Set the angle to the [-2 * pi, 2 * pi] interval
+        if not angle:
+            self._phase_angle = 0
```

Comment:
again I don't think this is necessary, no one should be able to call `gate.phase_angle = None`.